feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/presentation/staking/start/setupStakingType/SetupStakingTypeViewModel.kt

```diff
@@ -0,0 +1,196 @@
+
+    fun selectStakingType(stakingTypeRVItem: EditableStakingTypeRVItem, position: Int) {
+        if (stakingTypeRVItem.isSelected) return
+
+        launch {
+            val chainAsset = chainWithAsset.first().asset
+            val stakingTypeDetails = stakingTypesDataFlow.first()[position]
+                .stakingTypeDetails
+            val stakingTypeMixin = editingStakingTypeSelectionMixin.first()
+            val enteredAmount = stakingTypeMixin.enteredAmount() ?: return@launch
+
+            validationExecutor.requireValid(
+                validationSystem = stakingTypeMixin.getValidationSystem(stakingTypeDetails.stakingType) ?: return@launch,
+                payload = EditingStakingTypePayload(enteredAmount, stakingTypeDetails.stakingType, stakingTypeDetails.minStake),
+                validationFailureTransformer = { handleSetupStakingTypeValidationFailure(chainAsset, it, resourceManager) },
+            ) {
+                setRecommendedSelection(stakingTypeDetails.stakingType)
+            }
+        }
+    }
+
+    private fun setRecommendedSelection(stakingType: Chain.Asset.StakingType) {
+        launch {
+            editingStakingTypeSelectionMixin.first().setRecommendedSelection(stakingType)
+        }
+    }
+
+    private suspend fun mapStakingTypes(
+        asset: Asset,
+        stakingTypesDetails: List<EditableStakingType>,
+        selection: RecommendableMultiStakingSelection
+    ): List<EditableStakingTypeRVItem> {
+        return stakingTypesDetails.mapNotNull {
+            val stakingTarget = StakingTarget.Model(multiStakingSelectionFormatter.formatForStakingType(selection))
+            val selectedStakingType = selection.selection.stakingOption.stakingType
+            val stakingType = it.stakingTypeDetails.stakingType
+            when {
+                stakingType.isDirectStaking() -> EditableStakingTypeRVItem(
+                    isSelected = selectedStakingType.isDirectStaking(),
+                    isSelectable = it.isAvailable,
+                    title = resourceManager.getString(R.string.setup_staking_type_direct_staking),
+                    imageRes = R.drawable.ic_pool_staking_banner_picture,
+                    conditions = mapConditions(asset, it.stakingTypeDetails),
+                    stakingTarget = stakingTarget.takeIf { selectedStakingType.isDirectStaking() }
+                )
+                stakingType.isPoolStaking() -> EditableStakingTypeRVItem(
+                    isSelected = selectedStakingType.isPoolStaking(),
+                    isSelectable = it.isAvailable,
+                    title = resourceManager.getString(R.string.setup_staking_type_pool_staking),
+                    imageRes = R.drawable.ic_direct_staking_banner_picture,
+                    conditions = mapConditions(asset, it.stakingTypeDetails),
+                    stakingTarget = stakingTarget.takeIf { selectedStakingType.isPoolStaking() }
+                )
+                else -> null
+            }
```

Comment:
ViewModel should know about specific staking types used. Thats why we created separate interactors for each of them. Consider putting this code into a separe formatter like `MultiStakingSelectionFormatter`

Also it seems like the only piece that is depends on staking type is `title` and `imageRes`. Everything else is the same